### DISCUSSION TITLE
PHPSpreadsheet version not updated?

### DISCUSSION THREAD

**Discussion by toxyy** [pimcore/pimcore#17087] [Q&A / Support] [answered]
In the composer.json, PHPSpreadsheet is set to ^1.24, though the latest version on their github is version 2.1.0. Is there any reason that it hasn't been updated? I've tried looking around but can't find anything.

**Comment by kingjia90** [comment]
Thank you for reporting, there are no any particular and known reason (like sort of technical limitation) on why it hasn't updated, it's more like we didn't "track" their releases and weren't aware of the newest version.

Before adding support to any next major version, we usually needs some additional checks to see if the breaking changes on their end would cause breaking changes on our end but looking at https://github.com/pimcore/pimcore/pull/17091 and their changelog , it seems that would be fine.

Howere, i've found something more interesting, we don't even actually require that in the pimcore/pimcore bundle, the only usages found are in the admin ui bundle 
https://github.com/pimcore/admin-ui-classic-bundle/blob/93643ad21d5877b3d60973418414634f7cd3a62d/src/Helper/GridHelperService.php#L21-L23
and
https://github.com/pimcore/data-importer/blob/f25246e4a6537f47f72a600022886830b1cd30c2/src/DataSource/Interpreter/XlsxFileInterpreter.php#L18

so we may likely need some further refactoring, the main problem is that may be requiring a major release of the core, so for the moment the only safe move would be merging https://github.com/pimcore/pimcore/pull/17091 and move it out in pimcore 12, i have to bring this to the attention of the supervisor first to see if it's the right approach.
Will let you know

**Reply by kingjia90** [reply, marked as answer]
So the final decision is to remove from the core and move it to the bundles in the next minor releases, additionally we are extending the support to v2 as it seems not affecting our BC promises.
Please see the progression on https://github.com/pimcore/pimcore/issues/17092

Hope it helps solving your initial issue.
Thank you again for giving the input!

**Reply by toxyy** [reply]
Thanks for the quick responses!
